tidy: Add a rule ensuring that `//` comments are followed by a space in Rust

mrobinson opened the pull request:
This shows up sometimes in code reviews, so it makes sense that tidy
enforces it. `rustfmt` supports this via comment normalization, but it
does many other things and is still an unstable feature (with bugs).

Testing: There are new tidy tests for this change.

sagudev commented on support/crown/tests/compile-fail/missing_allow_in_rc.rs:
It does not: https://github.com/servo/servo/actions/runs/16988125011/job/48161379621#step:15:505

sagudev replied:
rustfmt uses this to detect comments that needs fixing: https://github.com/rust-lang/rustfmt/blob/0332da01486508710f2a542111e40513bfb215aa/src/comment.rs#L19

mrobinson committed:
tidy: Add a rule ensuring that `//` comments are followed by a space in Rust

This shows up sometimes in code reviews, so it makes sense that tidy
enforces it. `rustfmt` supports this via comment normalization, but it
does many other things and is still an unstable feature (with bugs).

Signed-off-by: Martin Robinson <mrobinson@igalia.com>

sagudev reviewed (changes requested):
We need tests for
- comments in same line as code
- compile-test comments
